pkg/pillar/cmd/usbmanager/usbmanager.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package usbmanager
+
+import (
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentbase"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/cmd/domainmgr"
+	"github.com/lf-edge/eve/pkg/pillar/hypervisor"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/utils"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	agentName = "usbmanager"
+	// Time limits for event loop handlers
+	errorTime            = 3 * time.Minute
+	warningTime          = 40 * time.Second
+	stillRunningInterval = 25 * time.Second
+)
+
+var (
+	logger *logrus.Logger
+	log    *base.LogObject
+)
+
+type usbdevice struct {
+	busnum                  uint16
+	portnum                 string
+	devnum                  uint16
+	vendorID                uint32
+	productID               uint32
+	devicetype              string
+	usbControllerPCIAddress string
+	ueventFilePath          string
+}
+
+type usbmanagerContext struct {
+	agentbase.AgentBase
+	SubAssignableAdapters pubsub.Subscription
+	SubDomainStatus       pubsub.Subscription
+	controller            usbmanagerController
+	runningDomains        map[string]struct{} // mapping DomainStatus.DomainName
+}
+
+func newUsbmanagerContext() *usbmanagerContext {
+	usbCtx := &usbmanagerContext{}
+	usbCtx.controller.init()
+
+	usbCtx.runningDomains = make(map[string]struct{})
+
+	return usbCtx
+}
+
+// Run - Main function - invoked from zedbox.go
+func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, arguments []string) int {
+	logger = loggerArg
+	log = logArg
+
+	usbCtx := newUsbmanagerContext()
+
+	agentbase.Init(usbCtx, logger, log, agentName,
+		agentbase.WithPidFile(),
+		agentbase.WithWatchdog(ps, warningTime, errorTime),
+		agentbase.WithArguments(arguments))
+
+	// Wait until we have been onboarded aka know our own UUID, but we don't use the UUID
+	err := utils.WaitForOnboarded(ps, log, agentName, warningTime, errorTime)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Functionf("processed onboarded")
+
+	if err := utils.WaitForVault(ps, log, agentName, warningTime, errorTime); err != nil {
+		log.Fatal(err)
+	}
+	log.Functionf("processed Vault Status")
+
+	var subs []pubsub.Subscription
+
+	currentHypervisor := domainmgr.CurrentHypervisor()
```

Comment:
Note that although all microservices currently run inside the same process (to decrease the EVE image size), they are using only pubsub to talk to each other, avoiding any global variables.
Having usbmanager retrieve hypervisor from domainmgr in this way means that it is effectively not a standalone microservice, rather an extension of domainmgr. It probably does not therefore make much sense to have it as a separate `pillar/cmd` package. Perhaps it could be a package directly under pillar that domainmgr uses?

Reply:
Does usbmanager actually need pointer to hypervisor or does it only need to determine if KVM or Xen or something else is being used?

Reply:
We have some packages under pillar which are only used from one place (like pillar/hypervisor). The idea is to simply avoid large packages.
But if you avoid direct dependency on domainmgr, then usbmanager could be seen as a standalone microservice and pillar/cmd is a good place.

Reply:
unfortunately it can be overriden with this flag: https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/domainmgr/domainmgr.go#L128

perhaps if I extract this method out of domainmgr and make an `init()` function that calls it, then it could be used by `domainmgr` and `usbmanager`.
But if we then later decide to split up the services into several processes, both services have to be started with the same flag.

Reply:
I added https://github.com/lf-edge/eve/pull/3443/commits/5cb908427719e05930402c1e253135e4b7aa62d6 but I am not really convinced ...